### ISSUE TITLE
Update support for rules

### ISSUE THREAD

**Issue by BrianSipos** [JHUAPL-DTNMA/dtnma-tools#78]
Add back TBR and SBR support with a rule processing thread.

This can be compile-time guarded eventually to disable `ietf-dtnma-agent:rules` feature.

**Comment by justinethier** [comment]
Breaking this ticket down into several PR's:

- Add TBR / SBR definitions
- Ability to execute TBR's
- Ability to execute SBR's
- ADM / ODM support for rules
- Cleanup (compile-time guards, etc)

There would also be a follow-on task to add rule support to CAMP.

**Comment by BrianSipos** [comment]
@justinethier has this ticket been fully addressed, or do more ADM function bodies need to be fully defined?

**Comment by justinethier** [comment]
@BrianSipos No, the rule CTRL's from https://github.com/JHUAPL-DTNMA/dtnma-adms/pull/15 need to be implemented. 

However those depend upon ODM PR (https://github.com/JHUAPL-DTNMA/dtnma-adms/pull/18) which needs to be implemented first. I am currently working on those.
